R/unneeded_concatenation_linter.R

```diff
@@ -5,7 +5,9 @@
 #' @param allow_single_expression Logical, default `TRUE`. If `FALSE`, one-expression
 #'   usages of `c()` are always linted, e.g. `c(x)` and `c(matrix(...))`. In some such
 #'   cases, `c()` is being used for its side-effect of stripping non-name attributes;
-#'   it is usually preferable to use [as.vector()] to accomplish the same more readably.
+#'   it is usually preferable to use the more readable [as.vector()] instead. But the
+#'   latter may not always be a suitable replacement (e.g., if `c(r6_object)`
+#'   is used to convert an `R6` class instance to a list).
```

Comment:
``` r
library(R6)

Person <- R6Class("Person", public = list("name" = NA))
Bob <- Person$new()

c(Bob)
#> [[1]]
#> <Person>
#>   Public:
#>     clone: function (deep = FALSE) 
#>     name: NA

as.vector(Bob)
#> Error in as.vector(x, mode): cannot coerce type 'environment' to vector of type 'any'
```

<sup>Created on 2022-07-28 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
The note could mention `list(env)` as a suitable replacement for `c(env)` explicitly.

Reply:
Thanks. Done!